src/mbedtls.c

```diff
@@ -758,94 +760,92 @@ bool HMAC_Validate(COSE_MacMessage * pcose, int HSize, int TSize, const byte * p
 	mbedtls_md_free(&contx); 
 	return false;
 }
-/*
 
 #define COSE_Key_EC_Curve -1
 #define COSE_Key_EC_X -2
 #define COSE_Key_EC_Y -3
 #define COSE_Key_EC_d -4
 
-EC_KEY * ECKey_From(const cn_cbor * pKey, int * cbGroup, cose_errback * perr)
+bool ECKey_From(const cn_cbor * pKey, mbedtls_ecp_keypair *keypair, cose_errback * perr)
 {
-	EC_KEY * pNewKey = EC_KEY_new();
-	byte  rgbKey[512+1];
+	byte  rgbKey[MBEDTLS_ECP_MAX_PT_LEN];
 	int cbKey;
+	int cbGroup;
 	const cn_cbor * p;
-	int nidGroup = -1;
-	EC_POINT * pPoint = NULL;
+	mbedtls_ecp_group_id groupId;
 
-	p = cn_cbor_mapget_int(pKey, COSE_Key_EC_Curve);
+	p = cn_cbor_mapget_int(pKey, COSE_Key_Type);
 	CHECK_CONDITION(p != NULL, COSE_ERR_INVALID_PARAMETER);
+	if(p->type == CN_CBOR_INT) {
+		CHECK_CONDITION(p->v.sint == COSE_Key_Type_EC2, COSE_ERR_INVALID_PARAMETER);
```

Comment:
Since you are checking for the value of EC2, this is not correct.  EC2 must be a UINT as the value is positive.